test/e2e-go/restAPI/simulate/simulateRestAPI_test.go

```diff
@@ -2695,3 +2695,86 @@ int 1
 	}
 	a.Equal(expectedResult, resp)
 }
+
+func TestSimulateWithFixSigners(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	a := require.New(fixtures.SynchronizedTest(t))
+	var localFixture fixtures.RestClientFixture
+	localFixture.Setup(t, filepath.Join("nettemplates", "TwoNodes50EachFuture.json"))
+	defer localFixture.Shutdown()
+
+	testClient := localFixture.LibGoalClient
+
+	_, err := testClient.WaitForRound(1)
+	a.NoError(err)
+
+	wh, err := testClient.GetUnencryptedWalletHandle()
+	a.NoError(err)
+	addresses, err := testClient.ListAddresses(wh)
+	a.NoError(err)
+	_, senderAddress := helper.GetMaxBalAddr(t, testClient, addresses)
+	if senderAddress == "" {
+		t.Error("no addr with funds")
+	}
+	a.NoError(err)
+
+	rekeyTxn, err := testClient.ConstructPayment(senderAddress, senderAddress, 0, 1, nil, "", [32]byte{}, 0, 0)
+	a.NoError(err)
+
+	var authAddr basics.Address
+	crypto.RandBytes(authAddr[:])
+	rekeyTxn.RekeyTo = authAddr
+
+	txn, err := testClient.ConstructPayment(senderAddress, senderAddress, 0, 1, nil, "", [32]byte{}, 0, 0)
+	a.NoError(err)
+
+	gid, err := testClient.GroupID([]transactions.Transaction{rekeyTxn, txn})
+	a.NoError(err)
+
+	rekeyTxn.Group = gid
+	txn.Group = gid
+
+	simulateRequest := v2.PreEncodedSimulateRequest{
+		TxnGroups: []v2.PreEncodedSimulateRequestTransactionGroup{
+			{
+				Txns: []transactions.SignedTxn{{Txn: rekeyTxn}, {Txn: txn}},
+			},
+		},
+		AllowEmptySignatures: true,
+		FixSigners:           true, // When this is true: HTTP 400 Bad Request: failed to decode object: json decode error [pos 1]: only encoded map or array can be decoded into a struct
```

Comment:
@jasonpaulos Any idea why this is throwing a JSON decode error? I've updated `PreEncodedSimulateRequest` with the proper codec tag, but the request still seems to be malformed for some reason.

Reply:
I'm not sure what decode error you're seeing. When I run this locally, I only see that the transaction failed with the wrong auth addr.

Although I did have to merge in master to get the code to compile, since we updated to go 1.21. I suggested doing the same, making sure you're on go 1.21 as well, and see if that helps.

I've opened this with my merge commit in case it's helpful: https://github.com/joe-p/go-algorand/pull/1

Reply:
Ah ok I wasn't rebuilding algod, thinking the test runner was doing it automatically, thus the encoding error. 

> When I run this locally, I only see that the transaction failed with the wrong auth addr.

I missed the `convert...` functions in the simulate handler. [e8cec95](https://github.com/algorand/go-algorand/pull/5942/commits/e8cec9520bedf6ca62012ebafaad63efe1f994de) fixed that and the request is properly working now. The response is still missing the `FixedSigner` in the `TxnResult` for some reason, but at least the request is working now.